aiormq/exceptions.py

```diff
@@ -136,9 +139,20 @@ class MethodNotImplemented(AMQPError):
 
 
 class DeliveryError(AMQPError):
-    __slots__ = 'message', 'frame'
+    __slots__ = 'returned_message', 'frame'
+
+    message = '%s: %s'
 
-    def __init__(self, message, frame):
-        self.message = message
+    def __init__(self, frame: spec.Frame,
+                 returned_message: DeliveredMessage = None):
         self.frame = frame
-        super().__init__()
+        self.returned_message = returned_message
+
+        if returned_message:
+            d = returned_message.delivery
```

Comment:
That really looks useful. But split this case to another inherited Exception will be better I guess.

Reply:
Do you suggest to use a child exception for the case when message is published with `mandatory` flag but can't be routed to any configured queue? I.e. we could name it `PublishError`.